File: Lexor/UI/lexor_shared/lib/src/contract_status.dart
enum ContractStatus {
  active(1, 'Aktivan'),
  expired(2, 'Istekao'),
  upcoming(3, 'Budući');

  final int code;
  final String label;

  const ContractStatus(this.code, this.label);

  static ContractStatus? fromCode(int? code) {
    for (final s in values) {
      if (s.code == code) return s;
    }
    return null;
  }
}

File: Lexor/Lexor.Model/Enums/ContractStatus.cs
namespace Lexor.Model.Enums
{
    // Derived from the contract's date range relative to today — never stored in the database.
    public enum ContractStatus
    {
        Active = 1,    // today is within [StartDate, EndDate] (or EndDate is open)
        Expired = 2,   // EndDate has already passed
        Upcoming = 3   // StartDate is still in the future
    }
}
